astroid/brain/brain_typing.py

```diff
@@ -179,33 +188,48 @@ def infer_typing_alias(
     res = next(node.args[0].infer(context=ctx))
```

Comment:
You didn't change it, but I noticed that the line above is redundant:
```py
if not isinstance(node, nodes.Call):
    return None
```
This will never match. `node` is always a `nodes.Call` instance.